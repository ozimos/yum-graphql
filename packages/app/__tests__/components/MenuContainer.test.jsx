/* global React:false, shallow:false toJson:false */

import { MenuContainer } from
  '../../src/components/orderCart/ConnectedMenuContainer';
import { menuMeals } from '../__mocks__/menuDataMock';

const props = {
  menu: menuMeals,
  menuError: '',
  dispatch: jest.fn(),
  notify: jest.fn(),
  postMenu: jest.fn(),
  clearMenu: jest.fn(),
};

describe('MenuContainer Component', () => {
  const setup = () => shallow(<MenuContainer {...props} />);

  it('should render without throwing an error', () => {
    const wrapper = setup();

    expect(wrapper
      .exists(<table className="table" />)).toBe(true);
  });

  it('renders correctly', () => {

    let wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();

    const propsWithExtraClass = { ...props, addClass: 'string' };
    const setupWithExtraClass = () => shallow(<MenuContainer
      {...propsWithExtraClass}
    />);
    wrapper = setupWithExtraClass();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call `placeOrder` when the place order button is clicked', () => {
    const wrapper = setup();

    const placeOrderSpy = jest.spyOn(
      wrapper.instance(),
      'placeOrder'
    );
    wrapper.instance().placeOrder();
    expect(placeOrderSpy).toHaveBeenCalled();
  });

  it('should call `placeOrder` when there is menu error' +
  ' and menu is not posted ', () => {
    const menuError = 'error';
    const propsWithMenuError = { ...props, menuError };
    const setupWithMenuError = () => shallow(<MenuContainer
      {...propsWithMenuError}
    />);

    const wrapper = setupWithMenuError();

    const placeOrderSpy = jest.spyOn(
      wrapper.instance(),
      'placeOrder'
    );
    wrapper.instance().placeOrder();
    expect(placeOrderSpy).toHaveBeenCalled();
  });

});

