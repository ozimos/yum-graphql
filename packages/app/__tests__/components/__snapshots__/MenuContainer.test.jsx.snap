// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuContainer Component renders correctly 1`] = `
<div
  className=""
>
  <div
    className="flexbox"
  >
    <h5>
      Menu Editor
    </h5>
    <button
      className="btn title-button"
    >
      ✖
    </button>
  </div>
  <div
    className="scroll"
  >
    <div
      className="flexbox"
      key="meal1"
    >
      <span
        className="text_left long_string"
        key="meal1"
      >
        Eba and Egusi
      </span>
      <button
        className="btn title-button"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
    <div
      className="flexbox"
      key="meal2"
    >
      <span
        className="text_left long_string"
        key="meal2"
      >
        Rice and Stew
      </span>
      <button
        className="btn title-button"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
  <div
    className="flexbox"
  >
    <button
      className="btn btn-cart"
      disabled={false}
      onClick={[Function]}
    >
      Post meals to Menu
    </button>
    <button
      className="btn btn-cart"
      disabled={false}
      onClick={[MockFunction]}
    >
      Clear Menu
    </button>
  </div>
</div>
`;

exports[`MenuContainer Component renders correctly 2`] = `
<div
  className="string"
>
  <div
    className="flexbox"
  >
    <h5>
      Menu Editor
    </h5>
    <button
      className="btn title-button"
    >
      ✖
    </button>
  </div>
  <div
    className="scroll"
  >
    <div
      className="flexbox"
      key="meal1"
    >
      <span
        className="text_left long_string"
        key="meal1"
      >
        Eba and Egusi
      </span>
      <button
        className="btn title-button"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
    <div
      className="flexbox"
      key="meal2"
    >
      <span
        className="text_left long_string"
        key="meal2"
      >
        Rice and Stew
      </span>
      <button
        className="btn title-button"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
  <div
    className="flexbox"
  >
    <button
      className="btn btn-cart"
      disabled={false}
      onClick={[Function]}
    >
      Post meals to Menu
    </button>
    <button
      className="btn btn-cart"
      disabled={false}
      onClick={[MockFunction]}
    >
      Clear Menu
    </button>
  </div>
</div>
`;
